### PR TITLE
fix email is sent when normal confirmation icon clicked

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#17584 ` email is sent when normal confirmation icon clicked`](http://community.openproject.org/work_packages/17584)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://0xf013-feature-17584-email-is-sent-when-normal-confirmation-ico.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: karma.conf.js [hunk 1]
```diff
@@ -38,7 +38,8 @@ module.exports = function(config) {
     client: {
       mocha: {
         ui: 'bdd'
-      }
+      },
+      captureConsole: true
```
Comment (myabc) [inline karma.conf.js]: @0xF013 does this belong in this commit?

**Comment by 0xF013** [comment]
@myabc It was added for commodity of debugging the karma tests I added. Do you think it belongs in a separate PR?

**Comment by myabc** [comment]
@0xF013 yes, as it's not really something related.

File: frontend/app/ui_components/inplace-editor-directive.js [hunk 2]
```diff
@@ -118,7 +118,7 @@ module.exports = function($timeout, InplaceEditorDispatcher) {
       var data = angular.copy($scope.entity.form.embedded.payload.props);
       InplaceEditorDispatcher.dispatchHook($scope, 'submit', data);
       $scope.isBusy = true;
-      var result = WorkPackageService.updateWorkPackage($scope.entity, data);
+      var result = WorkPackageService.updateWorkPackage($scope.entity, data, withEmail);
```
Comment (myabc) [inline frontend/app/ui_components/inplace-editor-directive.js]: @0xF013 can you call this argument `notify` too, for consistency.

File: frontend/tests/unit/tests/work_packages/directives/inplace-editor-directive-test.js [hunk 3]
```diff
@@ -211,13 +211,34 @@ describe('inplaceEditor Directive', function() {
         });
       });
       describe('submit', function() {
-        context('general', function() {
+        var emptyPromiseResponse = {
+          'then': $.noop,
```
Comment (myabc) [inline frontend/tests/unit/tests/work_packages/directives/inplace-editor-directive-test.js]: @0xF013 See 067ccd2b.

File: frontend/app/services/work-package-service.js [hunk 4]
```diff
@@ -139,9 +139,10 @@ module.exports = function($http, PathHelper, WorkPackagesHelper, HALAPIResource,
       });
     },
 
-    updateWorkPackage: function(workPackage, data) {
+    updateWorkPackage: function(workPackage, data, notify) {
       var options = { ajax: {
         method: 'PATCH',
+        url: workPackage.links.updateImmediately.href + '?notify=' + (notify ? 'true' : 'false'),
```
Comment (myabc) [inline frontend/app/services/work-package-service.js]: @0xF013 we could take advantage of URI.js here. Then we don't need to fiddle around with String concatenation and coercion: 

``` js
URI(workPackage.links.updateImmediately.href).addSearch('notify', notify).toString();
```

**Pull request merged**